src/trafficcontroller/dopplerproxy/doppler_proxy.go

```diff
@@ -137,13 +137,24 @@ func (proxy *Proxy) serveAppLogs(paths []string, writer http.ResponseWriter, req
 		return
 	}
 
-	authorized, err := proxy.logAuthorize(authToken, appId, proxy.logger)
+	status, _ := proxy.logAuthorize(authToken, appId, proxy.logger)
+	if status != http.StatusOK {
+		switch status {
+		case http.StatusUnauthorized:
+			writer.WriteHeader(status)
+			writer.Header().Set("WWW-Authenticate", "Basic")
+			fmt.Fprintf(writer, "You are not authorized.")
+		case http.StatusForbidden:
+			status = http.StatusNotFound
+			fallthrough
+		case http.StatusNotFound:
+			writer.WriteHeader(status)
+			fmt.Fprintf(writer, "App ID is not found.")
+		default:
+			writer.WriteHeader(status)
+			fmt.Fprintf(writer, "Internal server error occurred.")
+		}
 
-	if !authorized {
-		writer.Header().Set("WWW-Authenticate", "Basic")
-		writer.WriteHeader(http.StatusUnauthorized)
-		fmt.Fprintf(writer, "You are not authorized. %s", err.Error())
-		proxy.logger.Warnf("auth token not authorized to access appId [%s].", appId)
```

Comment:
It is already logged in the log_access_authorizer, but I can add the duplicate log line.